### PR TITLE
'Circle Retry' example

### PR DESCRIPTION
**Pull request by greg-1-anderson**
Add a Circle CI 'retry build' example, and a 'secrets' script to help manage external service secrets.

### CODE REVIEW

**Comment by joshkoenig** [comment]
Can we split out the `scripts/secrets` to it's own PR? I would love to get both of these things in, but they're really quite separate.

**Comment by greg-1-anderson** [comment]
I created a new PR, #14, containing only the 'secrets' script.

**Comment by joshkoenig** [comment]
I think un-DRYing the secrets handling would be good here, so we can follow our initial guideline of "one big script" for each example.

File: circle_retry/circle_retry.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+    else {
+      echo "Could not parse parameter file $circle_parameters_file:\n$parameter_contents\n";
+    }
+  }
+}
+
+// Load our hidden credentials.
+// See the README.md for instructions on storing secrets.
+$secrets = json_decode(file_get_contents($_SERVER['HOME'] . '/files/private/secrets.json'), 1);
```
Comment (joshkoenig) [inline circle_retry/circle_retry.php]: This example should have just one secrets/config json file, not two.

**Comment by joshkoenig** [comment]
I don't see how "retry" is used in practice. What it should do is trigger a CI test via circle, sending in the current tag in testing so that it can be used.

**Comment by greg-1-anderson** [comment]
The Circle API does not provide a way to trigger a new build; you can only retry an existing build. The retry API does not allow you to pass any parameters; it just runs the same build as last time. The retried build could be set up to always test whatever is active on a given Pantheon site and environment, though.

This is per https://circleci.com/docs/api#retry-build.  Perhaps there is some other integration mechanism that Github uses that we could hook into?  The problem with retrying a build is that you lose your history of passed / failed builds, the commit-hash of what was tested, etc.

**Comment by greg-1-anderson** [comment]
POST: /project/:username/:project/tree/:branch maybe?

**Comment by greg-1-anderson** [comment]
Hmmm: https://circleci.com/docs/parameterized-builds

**Comment by greg-1-anderson** [comment]
The problem here is that Circle is always testing Github branches / commit hashes.  What would work better would be some quicksilver-powered lite repo example that pushed commits back to the lite github repository. Then Circle builds could be triggered as usual. This is more complicated, of course.

Running behat directly on Pantheon would also be feasible; however, the big shortcoming here is in recording and reporting test results.  The existing services (Circle/Travis) are better at this.

**Comment by joshkoenig** [comment]
> Running behat directly on Pantheon would also be feasible; however, the big shortcoming here is in recording and reporting test results.

Definitely out of scope for the current state of quicksilver. 

I'd say we should explore the parameterized build option as an example.

**Comment by greg-1-anderson** [comment]
Okay, I'll fiddle with parameterized builds, and see how that works. The github repo would contain just the test code, not the Pantheon site. A parameter could indicate which environment to test.

**Comment by joshkoenig** [comment]
Closing this out. We'll return to external CI tools with fresh eyes later on.

**Pull request closed without merging**